src/importers/gcp/Dockerfile

```diff
@@ -0,0 +1,44 @@
+# The build image
+FROM python:3.12-slim-bookworm AS builder
+
+# Install and configure poetry
+RUN pip install poetry
+ENV POETRY_NO_INTERACTION=1 \
+    POETRY_VIRTUALENVS_IN_PROJECT=1 \
+    POETRY_VIRTUALENVS_CREATE=1 \
+    POETRY_CACHE_DIR=/tmp/poetry_cache
+
+WORKDIR /importers/gcp
+
+# Copy files needed to build
+COPY pyproject.toml poetry.lock ./
+RUN touch README.md
+
+# Install all dependencies
+RUN poetry install --without dev --no-root && rm -rf $POETRY_CACHE_DIR
+
+# The runtime image
+FROM python:3.12-slim-bookworm AS runtime
+
+# libmagic is needed for python-magic package
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    libmagic1 \
+  && rm -rf /var/lib/apt/lists/*
+
+# Set variables to point to the built virtualenv
+ENV VIRTUAL_ENV=/importers/gcp/.venv PATH="/importers/gcp/.venv/bin:$PATH"
+
+# Copy python virtualenv from the build step
+WORKDIR /importers/gcp
+COPY --from=builder ${VIRTUAL_ENV} ${VIRTUAL_ENV}
+
+# Install additional dependencies
+RUN pip install google-cloud-storage
```

Comment:
Why install these using pip directly? Shouldn't these dependencies be included in the pyproject.toml file since they are project-level dependencies?

I would opt to move these to the toml file, they could even be in an optional group to be used depending on the setup / configuration if gcp is required.

Reply:
Yeah, I agree, but it is a bit tricky for this one. It depends and hooks into the core codebase (directly talking to postgresq etc) and it would need to depend on the whole server package. We do not package up the server repo and there are no plans to do so as of now.

For now, I suggest we find a middle ground with managing the dependencies outside of the Docker file. I'll do that in a followup.

WDYT?

Reply:
Sounds reasonable to me.